File: web_dev_101/ruby_basics/euler3.rb
# The prime factors of 13195 are 5, 7, 13 and 29.

# What is the largest prime factor of the number 600851475143 ?
num = 1032

res = Array.new
primes = Array.new
i = 1
while i < num
    i_is_prime = true
    primes.each do |prime|
        i_is_prime = (i % prime == 0) ? false : true
    end

    if i_is_prime
        res = (num % i == 0) ? res + [i] : res
    end

    i = i + 1
end

print res

File: 2_web_dev_101/euler3.js
/*The prime factors of 13195 are 5, 7, 13 and 29.

What is the largest prime factor of the number 863 ?*/

var num = 672

primeArray = new Array();
res = new Array();

for (var test = 1; test < num; test++) {

	var test_is_prime = true;

	for (var i = 1; i < primeArray.length; i++) {
		if (test % primeArray[i] !== 0) {
			test_is_prime = false;
		}
	}

	if (test_is_prime && (num % test === 0)) {
		res.push(test)
	}
}

console.log(res.pop())